utilities/src/main/java/io/syndesis/qe/cucumber/MailFormatter.java

```diff
@@ -0,0 +1,145 @@
+package io.syndesis.qe.cucumber;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import cucumber.api.Result;
+import cucumber.api.event.EmbedEvent;
+import cucumber.api.event.EventListener;
+import cucumber.api.event.EventPublisher;
+import cucumber.api.event.TestCaseFinished;
+import cucumber.api.event.TestRunFinished;
+import cucumber.api.event.TestSourceRead;
+import gherkin.AstBuilder;
+import gherkin.Parser;
+import gherkin.ParserException;
+import gherkin.ast.Comment;
+import gherkin.ast.Feature;
+import gherkin.ast.GherkinDocument;
+import io.syndesis.qe.hooks.IssueHooks;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MailFormatter implements EventListener {
+
+    private static final Pattern SUSTAINER_PATTERN = Pattern.compile(".*@sustainer: ([a-zA-Z@.]+)");
+
+    private final String path;
+
+    ObjectMapper mapper = new ObjectMapper();
+
+    private Map<String, String> sustainers = new HashMap<>();
+    private Map<String, Feature> features = new HashMap<>();
+    private Map<String, List<IssueHooks.SimpleIssue>> scenarioIssues = new HashMap<>();
+
+    private StringBuilder reportBuilder = new StringBuilder();
+
+    public MailFormatter(String path) {
+        this.path = path;
+    }
+
+    @Override
+    public void setEventPublisher(EventPublisher publisher) {
+        publisher.registerHandlerFor(TestSourceRead.class, this::onTestSourceRead);
+        publisher.registerHandlerFor(TestCaseFinished.class, this::onTestCaseFinished);
+        publisher.registerHandlerFor(TestRunFinished.class, this::onTestRunFinished);
+        publisher.registerHandlerFor(EmbedEvent.class, this::onEmbed);
+    }
+
+    private void onEmbed(EmbedEvent t) {
```

Comment:
There's a bug here that test which passed after a failed test will have that failed test's issues bound to them, will fix